Type member widened to `Any` when passed out of inline method

## Minimized code

```Scala
class C {
  type A
  inline def test: A = ???
}
@main def m = {
  val c = C()
  val x: c.A = c.test
}
```

## Output

```
Found:    Any
Required: c.A
```

https://scastie.scala-lang.org/dB277yJnSs6qID8kkxvZxA

## Expectation

Compiles.

It now works with `transparent inline`. It also compiles OK as given with `inline`, but fails at Ycheck:
```
*** error while checking test.scala after phase inlining ***
exception occurred while compiling test.scala
java.lang.AssertionError: assertion failed: Found:    Any
Required: c.A
found: class Any in package scala with class Any, flags = open abstract <noinits> <permanent>, underlying = Any, {...}
expected: type A in class C with type A, flags = <deferred> <touched>, underlying = c.A, , Any, {...}
tree = {
  val C_this: (c : C) = c
  ??? :C_this.A
} while compiling test.scala
Exception in thread "main" java.lang.AssertionError: assertion failed: Found:    Any
Required: c.A
found: class Any in package scala with class Any, flags = open abstract <noinits> <permanent>, underlying = Any, {...}
expected: type A in class C with type A, flags = <deferred> <touched>, underlying = c.A, , Any, {...}
tree = {
  val C_this: (c : C) = c
  ??? :C_this.A
}
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
	at dotty.tools.dotc.transform.TreeChecker$Checker.adapt(TreeChecker.scala:598)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2960)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2964)
	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:309)```

--- a/tests/pos/i10343.scala
+++ b/tests/pos/i10343.scala
@@ -1,0 +1,11 @@
+//> using options -Ycheck:all
+
+class C {
+  type A
+  inline def test: A = ???
+}
+
+@main def m = {
+  val c = C()
+  val x: c.A = c.test
+}

